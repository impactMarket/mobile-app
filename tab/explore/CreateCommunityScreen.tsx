import React, { useState, useEffect } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import { TextInput, Card, Button, Paragraph } from 'react-native-paper';
import axios from 'axios';
import { connect, ConnectedProps } from 'react-redux';
import { IRootState } from '../../helpers/types';


interface INewCommunityFormFields {
    name: string;
    description: string;
    location: string;
    coverImage: string;
    claimAmount: string;
    frequency: string;
    timeIncrement: string;
    maxAmount: string;
}
interface ICommunityFormFieldsError {
    name: boolean;
    description: boolean;
    location: boolean;
    coverImage: boolean;
    claimAmount: boolean;
    frequency: boolean;
    timeIncrement: boolean;
    maxAmount: boolean;
}

const mapStateToProps = (state: IRootState) => {
    const { user, network } = state
    return { user, network }
};

const connector = connect(mapStateToProps)

type PropsFromRedux = ConnectedProps<typeof connector>

function CreateCommunityScreen(props: PropsFromRedux) {
    const [newCommunityForm, setNewCommunityForm] = useState<INewCommunityFormFields>(
        {
            name: '',
            description: '',
            location: '',
            coverImage: '',
            claimAmount: '',
            frequency: '',
            timeIncrement: '',
            maxAmount: '',
        }
    );
    const [communityFormError, setCommunityFormError] = useState<ICommunityFormFieldsError>(
        {
            name: false,
            description: false,
            location: false,
            coverImage: false,
            claimAmount: false,
            frequency: false,
            timeIncrement: false,
            maxAmount: false,
        }
    );

    useEffect(() => {

        axios.get('http://192.168.0.209:5000/api/community/all')
            .then(function (response) {
                // handle success
                console.log(response.data);
            })
            .catch(function (error) {
                // handle error
                console.log(error);
            })
            .finally(function () {
                // always executed
            });
    });

    const submitNewCommunity = () => {
        if (newCommunityForm.name.length === 0) {
            setCommunityFormError({ ...communityFormError, name: true });
        }
        else if (newCommunityForm.description.length === 0) {
            setCommunityFormError({ ...communityFormError, description: true });
        }
        else if (newCommunityForm.location.length === 0) {
            setCommunityFormError({ ...communityFormError, location: true });
        }
        else if (newCommunityForm.coverImage.length === 0) {
            setCommunityFormError({ ...communityFormError, coverImage: true });
        }
        else if (newCommunityForm.claimAmount.length === 0) {
            setCommunityFormError({ ...communityFormError, claimAmount: true });
        }
        else if (newCommunityForm.frequency.length === 0) {
            setCommunityFormError({ ...communityFormError, frequency: true });
        }
        else if (newCommunityForm.timeIncrement.length === 0) {
            setCommunityFormError({ ...communityFormError, timeIncrement: true });
        }
        else if (newCommunityForm.maxAmount.length === 0) {
            setCommunityFormError({ ...communityFormError, maxAmount: true });
        }
        else {
            // TODO: api call to save new community request
            axios.post('http://localhost:5000/api/community/add', {
                // TODO: change to contract address
                // deploy first
                walletAddress: props.user.celoInfo.address,
                name: newCommunityForm.name,
                description: newCommunityForm.description,
                location: {
                    title: newCommunityForm.location,
                    latitude: 5,
                    longitude: 6,
                },
                coverImage: newCommunityForm.coverImage,
            }, {
                headers: {
                    'Content-Type': 'application/json',
                    'Accept': 'application/json',
                }
            }).then((response: any) => {
                console.log(response);
            }).catch((error: any) => {
                console.log(error);
            });
        }
    }

    const handleTextInputChanges = (name: string, value: string) => {
        switch (name) {
            case 'name':
                setNewCommunityForm({ ...newCommunityForm, name: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, name: false });
                }
                break;
            case 'description':
                setNewCommunityForm({ ...newCommunityForm, description: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, description: false });
                }
                break;
            case 'location':
                setNewCommunityForm({ ...newCommunityForm, location: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, location: false });
                }
                break;
            case 'coverImage':
                setNewCommunityForm({ ...newCommunityForm, coverImage: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, coverImage: false });
                }
                break;
            case 'claimAmount':
                setNewCommunityForm({ ...newCommunityForm, claimAmount: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, claimAmount: false });
                }
                break;
            case 'frequency':
                setNewCommunityForm({ ...newCommunityForm, frequency: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, frequency: false });
                }
                break;
            case 'timeIncrement':
                setNewCommunityForm({ ...newCommunityForm, timeIncrement: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, timeIncrement: false });
                }
                break;
            case 'maxAmount':
                setNewCommunityForm({ ...newCommunityForm, maxAmount: value });
                if (value.length > 0) {
                    setCommunityFormError({ ...communityFormError, maxAmount: false });
                }
                break;
        }
    }

    return (
        <ScrollView>
            <Card>
                <Card.Cover
                    style={{ height: 100 }}
                    source={{ uri: 'https://picsum.photos/600' }}
                />
                <Card.Content style={styles.cardContent}>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        label="Name"
                        value={newCommunityForm.name}
                        onChangeText={value => handleTextInputChanges('name', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.name ? 'flex' : 'none'
                        }}
                    >
                        Name is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        label="Description"
                        value={newCommunityForm.description}
                        onChangeText={value => handleTextInputChanges('description', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.description ? 'flex' : 'none'
                        }}
                    >
                        Description is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        label="Location"
                        value={newCommunityForm.location}
                        onChangeText={value => handleTextInputChanges('location', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.location ? 'flex' : 'none'
                        }}
                    >
                        Location is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        label="Cover Image"
                        value={newCommunityForm.coverImage}
                        onChangeText={value => handleTextInputChanges('coverImage', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.coverImage ? 'flex' : 'none'
                        }}
                    >
                        Cover Image is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        keyboardType="numeric"
                        label="Claim Amount"
                        value={newCommunityForm.claimAmount}
                        onChangeText={value => handleTextInputChanges('claimAmount', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.claimAmount ? 'flex' : 'none'
                        }}
                    >
                        Claim Amount is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        keyboardType="numeric"
                        label="Frequency"
                        value={newCommunityForm.frequency}
                        onChangeText={value => handleTextInputChanges('frequency', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.frequency ? 'flex' : 'none'
                        }}
                    >
                        Frequency is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        keyboardType="numeric"
                        label="Time Increment"
                        value={newCommunityForm.timeIncrement}
                        onChangeText={value => handleTextInputChanges('timeIncrement', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.timeIncrement ? 'flex' : 'none'
                        }}
                    >
                        Time Increment is required!
                    </Paragraph>
                    <TextInput
                        style={styles.inputTextField}
                        mode="outlined"
                        keyboardType="numeric"
                        label="Max Amount"
                        value={newCommunityForm.maxAmount}
                        onChangeText={value => handleTextInputChanges('maxAmount', value)}
                    />
                    <Paragraph
                        style={{
                            color: 'red',
                            display: communityFormError.maxAmount ? 'flex' : 'none'
                        }}
                    >
                        Max Amount is required!
                    </Paragraph>
                    <Button
                        style={styles.inputTextField}
                        mode="contained"
                        onPress={() => submitNewCommunity()}
                    >
                        Create Community
                    </Button>
                </Card.Content>
            </Card>
        </ScrollView>
    );
}

const styles = StyleSheet.create({
    cardContent: {
        marginLeft: 30,
        marginRight: 30,
    },
    inputTextField: {
        marginTop: 10,
    },
});

export default connector(CreateCommunityScreen);