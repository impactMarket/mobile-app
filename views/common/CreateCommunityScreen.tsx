import React, { useState, useEffect } from 'react';
import {
    StyleSheet,
    ScrollView,
    Alert,
    Text,
    View,
    Picker,
    ImageBackground,
} from 'react-native';
import {
    Card,
    Button,
    Paragraph,
    Headline,
    Divider
} from 'react-native-paper';
import {
    connect,
    ConnectedProps
} from 'react-redux';
import {
    IRootState,
    ICommunityInfo,
    IUserState
} from '../../helpers/types';
import {
    requestCreateCommunity,
    celoWalletRequest
} from '../../services';
import { useNavigation } from '@react-navigation/native';
import * as Location from 'expo-location';
import config from '../../config';
import BigNumber from 'bignumber.js';
import ValidatedTextInput from '../../components/ValidatedTextInput';
import {
    humanifyNumber,
    loadContracts,
    validateEmail,
    getUserCurrencySymbol,
    amountToUserCurrency
} from '../../helpers';
import Header from '../../components/Header';
import {
    editCommunity,
    uploadImageAsync
} from '../../services/api';
import * as ImagePicker from 'expo-image-picker';
import i18n from '../../assets/i18n';


interface ICreateCommunityScreen {
    route: {
        params: {
            community: ICommunityInfo,
            user: IUserState,
        }
    }
}
const mapStateToProps = (state: IRootState) => {
    const { user, network } = state
    return { user, network }
};
const connector = connect(mapStateToProps)
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & ICreateCommunityScreen;

function CreateCommunityScreen(props: Props) {
    const navigation = useNavigation();

    const [editing, setEditing] = useState(false);
    const [sending, setSending] = useState(false);
    const [gpsLocation, setGpsLocation] = useState<Location.LocationData>();
    //
    const [isNameValid, setIsNameValid] = useState(false);
    const [isDescriptionValid, setIsDescriptionValid] = useState(false);
    const [isCityValid, setIsCityValid] = useState(false);
    const [isCountryValid, setIsCountryValid] = useState(false);
    const [isEmailValid, setIsEmailValid] = useState(true); // avoid initial error TODO: fix!
    const [isClaimAmountValid, setIsClaimAmountValid] = useState(false);
    const [isIncrementalIntervalValid, setIsIncrementalIntervalValid] = useState(false);
    const [isMaxClaimValid, setIsMaxClaimValid] = useState(false);
    //
    const [name, setName] = useState('');
    const [description, setDescription] = useState('');
    const [city, setCity] = useState('');
    const [country, setCountry] = useState('');
    const [email, setEmail] = useState('');
    const [coverImage, setCoverImage] = useState('');
    const [claimAmount, setClaimAmount] = useState('');
    const [baseInterval, setBaseInterval] = useState('86400');
    const [incrementInterval, setIncrementalInterval] = useState('');
    const [maxClaim, setMaxClaim] = useState('');
    const [visibility, setVisivility] = useState('public');

    useEffect(() => {
        if (props.route.params !== undefined) {
            const community = props.route.params.community as ICommunityInfo;
            if (community !== undefined) {
                setName(community.name);
                setDescription(community.description);
                setCity(community.city);
                setCountry(community.country);
                setEmail(community.email);
                setVisivility(community.visibility);
                setGpsLocation({
                    coords: {
                        latitude: community.gps.latitude,
                        longitude: community.gps.longitude,
                    }
                } as Location.LocationData)
                // cover image
                setClaimAmount(humanifyNumber(community.vars._claimAmount).toString());
                setBaseInterval(community.vars._baseInterval);
                setIncrementalInterval(new BigNumber(community.vars._incrementInterval).div(60).toString());
                setMaxClaim(humanifyNumber(community.vars._maxClaim).toString());
                // currency

                setIsNameValid(true);
                setIsDescriptionValid(true);
                setIsCityValid(true);
                setIsClaimAmountValid(true);
                setIsIncrementalIntervalValid(true);
                setIsMaxClaimValid(true);

                setEditing(true);
            }
        }
    }, []);

    useEffect(() => {
        if (props.network.community !== undefined && sending === true) {
            // wait until community is not undefined!
            setSending(false);
            navigation.goBack();
            Alert.alert(
                i18n.t('success'),
                i18n.t('requestNewCommunityPlaced'),
                [{ text: 'OK' }], { cancelable: false }
            );
        }
    }, [props.network.community]);

    const submitNewCommunity = async () => {
        if (gpsLocation === undefined) {
            // TODO: show error!
            return;
        }
        if (new BigNumber(maxClaim).lt(claimAmount)) {
            Alert.alert(
                i18n.t('failure'),
                i18n.t('claimBiggerThanMax'),
                [{ text: 'OK' }], { cancelable: false }
            );
            return;
        }
        if (new BigNumber(claimAmount).eq(0)) {
            Alert.alert(
                i18n.t('failure'),
                i18n.t('claimNotZero'),
                [{ text: 'OK' }], { cancelable: false }
            );
            return;
        }
        if (new BigNumber(maxClaim).eq(0)) {
            Alert.alert(
                i18n.t('failure'),
                i18n.t('maxNotZero'),
                [{ text: 'OK' }], { cancelable: false }
            );
            return;
        }
        setSending(true);
        const decimals = new BigNumber(10).pow(config.cUSDDecimals);
        if (editing) {
            const community = props.route.params.community as ICommunityInfo;
            const {
                _claimAmount,
                _baseInterval,
                _maxClaim,
                _incrementInterval
            } = props.route.params.community.vars;
            try {
                if (
                    !new BigNumber(claimAmount).multipliedBy(decimals).eq(_claimAmount) ||
                    baseInterval !== _baseInterval ||
                    parseInt(incrementInterval, 10) * 3600 !== parseInt(_incrementInterval, 10) ||
                    !new BigNumber(maxClaim).multipliedBy(decimals).eq(_maxClaim)
                ) {
                    // if one of the fields is changed, sent contract edit!
                    await celoWalletRequest(
                        props.user.celoInfo.address,
                        community.contractAddress,
                        await props.network.contracts.communityContract.methods.edit(
                            new BigNumber(claimAmount).multipliedBy(decimals).toString(),
                            new BigNumber(maxClaim).multipliedBy(decimals).toString(),
                            baseInterval,
                            (parseInt(incrementInterval, 10) * 60).toString(),
                        ),
                        'editcommunity',
                        props.network,
                    )
                }
                const success = await editCommunity(
                    community.publicId,
                    name,
                    description,
                    city,
                    country,
                    {
                        latitude: gpsLocation!.coords.latitude,
                        longitude: gpsLocation!.coords.longitude,
                    },
                    email,
                    visibility,
                    coverImage
                )
                if (!success) {
                    throw new Error('Some error!');
                } else {
                    await loadContracts(props.user.celoInfo.address, props.network.kit, props);
                    navigation.goBack();
                    Alert.alert(
                        i18n.t('success'),
                        i18n.t('communityUpdated'),
                        [{ text: 'OK' }], { cancelable: false }
                    );
                }
            } catch (e) {
                Alert.alert(
                    i18n.t('failure'),
                    i18n.t('errorUpdatingCommunity'),
                    [{ text: 'OK' }], { cancelable: false }
                );
            } finally {
                setSending(false);
            }
        } else {
            let uploadResponse, uploadImagePath;
            try {
                uploadResponse = await uploadImageAsync(coverImage);
                if (uploadResponse?.status === 200) {
                    uploadImagePath = uploadResponse.data.location;
                }
            } catch (e) {
                // log({ uploadResponse });
                // log({ uploadImagePath });
                // log({ e });
                Alert.alert(
                    i18n.t('failure'),
                    i18n.t('errorCreatingCommunity'),
                    [{ text: 'OK' }], { cancelable: false }
                );
                setSending(false);
                return;
            }
            requestCreateCommunity(
                props.user.celoInfo.address,
                name,
                description,
                city,
                country,
                {
                    latitude: gpsLocation!.coords.latitude,
                    longitude: gpsLocation!.coords.longitude,
                },
                email,
                visibility,
                uploadImagePath,
                {
                    claimAmount: new BigNumber(claimAmount).multipliedBy(decimals).toString(),
                    maxClaim: new BigNumber(maxClaim).multipliedBy(decimals).toString(),
                    baseInterval: baseInterval,
                    incrementInterval: (parseInt(incrementInterval, 10) * 60).toString(),
                },
            ).then((success) => {
                if (success) {
                    loadContracts(props.user.celoInfo.address, props.network.kit, props);
                    // the remaining process is done in useEffect
                } else {
                    Alert.alert(
                        i18n.t('failure'),
                        i18n.t('errorCreatingCommunity'),
                        [{ text: 'OK' }], { cancelable: false }
                    );
                    setSending(false);
                }
            });
        }
    }

    const enableGPSLocation = async () => {
        let { status } = await Location.requestPermissionsAsync();
        if (status !== 'granted') {
            // TODO: do some stuff
            return;
        }

        let loc = await Location.getCurrentPositionAsync();
        setGpsLocation(loc);
    }

    const pickImage = async () => {
        let result = await ImagePicker.launchImageLibraryAsync({
            mediaTypes: ImagePicker.MediaTypeOptions.Images,
            allowsEditing: true,
            aspect: [1, 1],
            quality: 1,
        });

        if (!result.cancelled) {
            setCoverImage(result.uri);
        }
    };

    const isSubmitAvailable = isNameValid &&
        isDescriptionValid &&
        isCityValid &&
        isCountryValid &&
        isEmailValid && email.length > 0 &&
        isClaimAmountValid &&
        isIncrementalIntervalValid &&
        isMaxClaimValid &&
        gpsLocation !== undefined &&
        coverImage.length > 0 &&
        !sending;

    if (props.user.celoInfo.address.length === 0) {
        return <View>
            <Header
                title={i18n.t('create')}
                navigation={navigation}
                hasBack={true}
            />
            <View style={styles.container}>
                <Text>
                    {i18n.t('needLoginToCreateCommunity')}
                </Text>
            </View>
        </View>
    }

    return (
        <>
            <Header
                title={editing ? i18n.t('edit') : i18n.t('create')}
                navigation={navigation}
                hasBack={true}
            >
                <Button
                    mode="text"
                    loading={sending}
                    disabled={!isSubmitAvailable}
                    onPress={() => submitNewCommunity()}
                >
                    {i18n.t('submit')}
                </Button>
            </Header>
            <ScrollView>
                <View style={styles.container}>
                    <Card elevation={8}>
                        <Card.Content style={{ margin: -16 }}>
                            <Headline style={styles.communityDetailsHeadline} >
                                {i18n.t('communityDetails').toUpperCase()}
                            </Headline>
                            <Text style={styles.createCommunityDescription}>
                                {i18n.t('createCommunityDescription')}
                            </Text>
                            <View>
                                <ImageBackground
                                    source={coverImage.length === 0 ? require('../../assets/images/placeholder.png') : { uri: coverImage }}
                                    style={styles.imageCover}
                                >
                                    <Button
                                        mode="contained"
                                        style={{ margin: 16 }}
                                        onPress={pickImage}
                                    >
                                        {coverImage.length === 0 ? i18n.t('selectCoverImage') : i18n.t('changeCoverImage')}
                                    </Button>
                                </ImageBackground>
                                <ValidatedTextInput
                                    label={i18n.t('communityName')}
                                    marginBox={16}
                                    value={name}
                                    maxLength={32}
                                    required={true}
                                    setValid={setIsNameValid}
                                    onChangeText={value => setName(value)}
                                />
                                <Divider />
                                <ValidatedTextInput
                                    label={i18n.t('shortDescription')}
                                    marginBox={16}
                                    value={description}
                                    maxLength={512}
                                    required={true}
                                    setValid={setIsDescriptionValid}
                                    onChangeText={value => setDescription(value)}
                                    multiline={true}
                                    numberOfLines={6}
                                />
                                <Divider />
                                <ValidatedTextInput
                                    label={i18n.t('city')}
                                    marginBox={16}
                                    value={city}
                                    maxLength={32}
                                    required={true}
                                    setValid={setIsCityValid}
                                    onChangeText={value => setCity(value)}
                                />
                                <Divider />
                                <ValidatedTextInput
                                    label={i18n.t('country')}
                                    marginBox={16}
                                    value={country}
                                    maxLength={32}
                                    required={true}
                                    setValid={setIsCountryValid}
                                    onChangeText={value => setCountry(value)}
                                />
                                {gpsLocation === undefined && <Button
                                    mode="outlined"
                                    style={{ marginHorizontal: 16 }}
                                    onPress={() => enableGPSLocation()}
                                >
                                    {i18n.t('getGPSLocation')}
                                </Button>}
                                {gpsLocation !== undefined && <Button
                                    icon="check"
                                    mode="outlined"
                                    style={{ marginHorizontal: 16 }}
                                    disabled={true}
                                >
                                    {i18n.t('validCoordinates')}
                                </Button>}
                                <ValidatedTextInput
                                    label={i18n.t('email')}
                                    marginBox={16}
                                    value={email}
                                    maxLength={32}
                                    required={true}
                                    keyboardType="email-address"
                                    isValid={isEmailValid}
                                    whenEndEditing={(e) => setIsEmailValid(validateEmail(email))}
                                    onChangeText={value => setEmail(value)}
                                />
                            </View>
                        </Card.Content>
                    </Card>
                    <Headline style={styles.contractDetailsHeadline}>
                        {i18n.t('contractDetails')}
                    </Headline>
                    <View style={{ marginBottom: 15 }}>
                        <View>
                            <ValidatedTextInput
                                label={i18n.t('claimAmount')}
                                placeholder="$0"
                                marginBox={10}
                                keyboardType="numeric"
                                value={claimAmount}
                                required={true}
                                setValid={setIsClaimAmountValid}
                                onChangeText={value => setClaimAmount(value)}
                            />
                            {
                                claimAmount.length > 0 && <Text
                                    style={styles.aroundCurrencyValue}
                                >
                                    {i18n.t('aroundValue', {
                                        symbol: getUserCurrencySymbol(props.user.user),
                                        amount: amountToUserCurrency(
                                            new BigNumber(claimAmount)
                                                .multipliedBy(new BigNumber(10).pow(config.cUSDDecimals)),
                                            props.user.user
                                        )
                                    })}
                                </Text>
                            }
                        </View>
                        <Divider />
                        <View>
                            <ValidatedTextInput
                                label={i18n.t('totalClaimPerBeneficiary')}
                                placeholder="$0"
                                marginBox={10}
                                keyboardType="numeric"
                                value={maxClaim}
                                required={true}
                                setValid={setIsMaxClaimValid}
                                onChangeText={value => setMaxClaim(value)}
                            />
                            {
                                maxClaim.length > 0 && <Text
                                    style={styles.aroundCurrencyValue}
                                >
                                    {i18n.t('aroundValue', {
                                        symbol: getUserCurrencySymbol(props.user.user),
                                        amount: amountToUserCurrency(
                                            new BigNumber(maxClaim)
                                                .multipliedBy(new BigNumber(10).pow(config.cUSDDecimals)),
                                            props.user.user
                                        )
                                    })}
                                </Text>
                            }
                        </View>
                        <Divider />
                        <Paragraph style={styles.inputTextFieldLabel}>
                            {i18n.t('frequency')}
                        </Paragraph>
                        <View style={styles.pickerBorder}>
                            <Picker
                                selectedValue={baseInterval}
                                style={styles.picker}
                                onValueChange={(value) => setBaseInterval(value)}
                            >
                                <Picker.Item label={i18n.t('hourly')} value="3601" />
                                <Picker.Item label={i18n.t('daily')} value="86400" />
                                <Picker.Item label={i18n.t('weekly')} value="604800" />
                            </Picker>
                        </View>
                        <ValidatedTextInput
                            label={i18n.t('timeIncrementAfterClaim')}
                            marginBox={10}
                            keyboardType="numeric"
                            value={incrementInterval}
                            required={true}
                            setValid={setIsIncrementalIntervalValid}
                            onChangeText={value => setIncrementalInterval(value)}
                        />
                        <Divider />
                        <Paragraph style={styles.inputTextFieldLabel}>
                            {i18n.t('visibility')}
                        </Paragraph>
                        <View style={styles.pickerBorder}>
                            <Picker
                                selectedValue={visibility}
                                style={styles.picker}
                                onValueChange={(text) => setVisivility(text)}
                            >
                                <Picker.Item label={i18n.t('public')} value="public" />
                                <Picker.Item label={i18n.t('private')} value="private" />
                            </Picker>
                        </View>
                    </View>
                    <Text style={{ ...styles.textNote, marginVertical: 20 }}>
                        {i18n.t('createCommunityNote1')}
                    </Text>
                    <Text style={styles.textNote}>
                        {i18n.t('createCommunityNote2')}
                    </Text>
                </View>
            </ScrollView>
        </>
    );
}

const styles = StyleSheet.create({
    cardContent: {
        marginLeft: 30,
        marginRight: 30,
    },
    inputTextFieldLabel: {
        marginHorizontal: 10,
        color: 'grey',
        fontSize: 15,
        fontFamily: 'Gelion-Regular'
    },
    inputTextField: {
        padding: 10,
        marginVertical: 5,
        borderStyle: 'solid',
        borderColor: 'grey',
        borderWidth: 1,
        borderRadius: 5
    },
    container: {
        margin: 20
    },
    //
    textNote: {
        // color: 'grey',
        fontFamily: 'Gelion-Regular',
    },
    communityName: {
        fontSize: 25,
        fontWeight: 'bold',
        fontFamily: 'Gelion-Bold',
        color: 'white'
    },
    communityLocation: {
        fontSize: 20,
        color: 'white'
    },
    picker: {
        height: 50,
        width: '100%',
        fontFamily: 'Gelion-Regular',
    },
    pickerBorder: {
        margin: 10,
        borderStyle: 'solid',
        borderColor: 'grey',
        borderWidth: 1,
        borderRadius: 5
    },
    imageCover: {
        flex: 1,
        flexDirection: 'column',
        resizeMode: 'cover',
        justifyContent: "flex-end",
        width: '100%',
        height: 180,
    },
    aroundCurrencyValue: {
        position: 'absolute',
        marginHorizontal: 10,
        marginVertical: 50,
        right: 0,
        fontFamily: "Gelion-Regular",
        fontSize: 15,
        fontWeight: "normal",
        fontStyle: "normal",
        lineHeight: 15,
        letterSpacing: 0.25,
        color: "#7e8da6"
    },
    communityDetailsHeadline: {
        opacity: 0.48,
        fontFamily: "Gelion-Regular",
        fontSize: 13,
        fontWeight: "500",
        fontStyle: "normal",
        lineHeight: 12,
        letterSpacing: 0.7,
        paddingHorizontal: 16,
        paddingVertical: 10
    },
    contractDetailsHeadline: {
        opacity: 0.48,
        fontFamily: "Gelion-Regular",
        fontSize: 13,
        fontWeight: "500",
        fontStyle: "normal",
        lineHeight: 13,
        letterSpacing: 0.7,
        marginTop: 20,
        marginHorizontal: 10
    },
    createCommunityDescription: {
        fontFamily: 'Gelion-Regular',
        backgroundColor: '#f0f0f0',
        padding: 16,
        borderTopColor: '#d8d8d8',
        borderTopWidth: 1,
        borderBottomColor: '#d8d8d8',
        borderBottomWidth: 1
    }
});

export default connector(CreateCommunityScreen);