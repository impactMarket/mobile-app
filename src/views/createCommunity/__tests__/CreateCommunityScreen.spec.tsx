import { assert } from 'chai';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import CreateCommunityScreen from '../index';

const mockStore = configureMockStore();
const store = mockStore({});

describe('CreateCommunity screen test suite', () => {
    let screen: ShallowWrapper<any>;

    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('should render CreateCommunity screen correctly', () => {
        givenScreen();
        thenItRenderProperly();
    });

    function givenScreen() {
        assert.isDefined(CreateCommunityScreen);

        screen = shallow(
            <Provider store={store}>
                <CreateCommunityScreen />
            </Provider>
        );
    }

    function thenItRenderProperly() {
        expect(screen).toMatchSnapshot();
    }
});
