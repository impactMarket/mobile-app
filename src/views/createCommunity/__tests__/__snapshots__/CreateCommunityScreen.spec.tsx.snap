// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCommunity screen test suite should render CreateCommunity screen correctly 1`] = `ShallowWrapper {}`;
